src/ci_provider/buildkite/provider.rs

```diff
@@ -0,0 +1,336 @@
+use std::env;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use crate::{
+    ci_provider::provider::{CIProvider, CIProviderDetector},
+    config::Config,
+    helpers::get_env_variable,
+    prelude::*,
+    uploader::{RunEvent, Runner, UploadMetadata},
+    VERSION,
+};
+
+use super::logger::BuildkiteLogger;
+
+#[derive(Debug)]
+pub struct BuildkiteProvider {
+    pub owner: String,
+    pub repository: String,
+    pub ref_: String,
+    pub head_ref: Option<String>,
+    pub base_ref: Option<String>,
+    pub commit_hash: String,
+    pub event: RunEvent,
+    pub repository_root_path: String,
+}
+
+lazy_static! {
+    static ref GITHUB_URL_REGEX: Regex = Regex::new(
+        r"(?x)
+            (?:https://github.com/|git@github.com:)
+            (?P<owner>[^/]+)/(?P<repository>[^/.]+)\.git
+        "
+    )
+    .expect("Failed to compile GitHub URL regex");
+}
+
+impl BuildkiteProvider {
```

Comment:
Is it necessary to bind those functions to the BuildkiteProvider Struct? They could live outside since all of then (almost) are static functions